### DISCUSSION TITLE
does tooltip have movable ability ?

### DISCUSSION THREAD

**Discussion by fseyedsafar** [skydoves/Balloon#262] [Q&A] [answered]
Is there a way to stick the tool tip to the icon? 
For example, when the keyboard mode is changing, the tool tip in the bottom toolbar will also change its position !

**Comment by skydoves** [comment, marked as answer]
Probably you can get a solution from this #238

**Reply by fseyedsafar** [reply]
@skydoves 
Actually I tried to use it but I think my problem is different .

Consider that I have bottom toolbar and I show my tooltip on it . when the keyboard is displayed at first, my toolbar is placed on top of the keyboard and tooltip is shown on the toolbar and which is fine but when I close the keyboard, my bottom toolbar return to its place and the tooltip detaches from it . I mean my tooltip stays in its first place and is no longer displayed on the toolbar.
I'm looking for an ability to stick the tooltip to a component . In this case whenever the location of component changes the new location of tooltip will change .

Is there any way to solve my problem ?
Maybe you have fixed it before but I couldn't find the solution in your codes .

**Reply by skydoves** [reply]
Thanks for your details!

Unfortunately, we can't observe the location of the `PopupWindow` or the anchor (internal structure of the Balloon) on the Android platform yet, however here is a tricky solution: Instead, we can observe the visibility of the keyboard using the insets at least, then we can update the location of the `PopupWindow` using the `update(anchor)` method.

```kotlin
observeKeyboardVisibility {
  balloon.update(anchor)
}
```

**Reply by fseyedsafar** [reply]
ommm.... sorry again .
 `observeKeyboardVisibility ` is in your internal classes or I have to implement it myself ?

**Reply by skydoves** [reply]
No, that is an example method. I think there are a lot of keyboard visibility observation libraries.

**Reply by fseyedsafar** [reply]
Yes thanks a lot .
